### PR TITLE
Rspec

### PR DESCRIPTION
**Pull request by Enanodr**
## Summary

- Model test were added
- Article controller and Welcome controller were tested so far
- Gems used: rspec-rails, shoulda-matchers, factory_girl_rails and factory_girl_rspec

## TrelloCard
https://trello.com/c/42HVBgsr

### CODE REVIEW

File: blog/config/application.rb [hunk 1]
```diff
@@ -8,6 +8,15 @@
 
 module Blog
   class Application < Rails::Application
+    ActionDispatch::Callbacks.after do
```
Comment (epintos) [inline blog/config/application.rb]: Check this

File: blog/spec/controllers/articles_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,97 @@
+require 'rails_helper'
+
+RSpec.describe ArticlesController, type: :controller do
+  context 'sign in user' do
+    let!(:articles) { create_list(:article, 2) }
+    let!(:valid_article_edit) { Article.new(id: articles.first.id, title: 'some valid title', text: 'some text') }
```
Comment (epintos) [inline blog/spec/controllers/articles_controller_spec.rb]:
```ruby
let!(:invalid_article_edit) { build(:article, id: articles.first.id, title: 'shrt', text: 'some text') }
let!(:form_user) { build(:form_user) }
```

File: blog/spec/controllers/articles_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    let!(:invalid_article_edit) { Article.new(id: articles.first.id, title: 'shrt', text: 'some text') }
+    before do
+      user = FactoryGirl.build(:form_user)
+      user.save
+      sign_in user
+    end
+    describe 'GET #index' do
```
Comment (epintos) [inline blog/spec/controllers/articles_controller_spec.rb]:
``` ruby
describe 'GET #index' do

  context 'When the user is signed in' ...

  context "When the users is not signed in'

end
```

File: blog/spec/factories/articles.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+FactoryGirl.define do
+  factory :article do
+    title 'a title'
```
Comment (epintos) [inline blog/spec/factories/articles.rb]: Use faker

File: blog/spec/factories/form_users.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+FactoryGirl.define do
+  factory :form_user do
+    email 'example@wolox.com.ar'
+    password 'a password'
```
Comment (epintos) [inline blog/spec/factories/form_users.rb]: Faker

File: blog/spec/factories/users.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+FactoryGirl.define do
+  factory :formUser do
+    email 'example@wolox.com.ar'
+    encrypted_password 'an encrypted password'
```
Comment (epintos) [inline blog/spec/factories/users.rb]: Faker

File: blog/spec/models/comment_spec.rb [hunk 7]
```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  before(:each) do
+    @article = FactoryGirl.create(:article)
+    @comment = FactoryGirl.build(:comment, commenter: 'somebody', body: 'an interesting comment', article: @article)
```
Comment (epintos) [inline blog/spec/models/comment_spec.rb]: Use lets

File: blog/spec/models/comment_spec.rb [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+  end
+  it { expect(@comment).to be_a_new(Comment).with(commenter: 'somebody', body: 'an interesting comment') }
+
+  it { expect(@comment).to be_valid }
+
+  it 'belongs to an article' do
+    expect(@comment.article.title).to eq @article.title
+  end
+
+  # Should have always an article
```
Comment (epintos) [inline blog/spec/models/comment_spec.rb]: `# TODO: `

File: blog/spec/models/identity_spec.rb [hunk 9]
```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.describe Identity, type: :model do
+  before(:each) do
+    @user_created = FactoryGirl.build(:form_user)
```
Comment (epintos) [inline blog/spec/models/identity_spec.rb]: Use lets

File: blog/spec/controllers/welcome_controller_spec.rb [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+      end
+      it { is_expected.to respond_with :ok }
+      it { is_expected.to render_with_layout :application }
+      it 'should load articles' do
+        expect(assigns(:articles)).to eq articles
+      end
+      # article.errors.to be present
+    end
+
+    #TODO: test email
```
Comment (epintos) [inline blog/spec/controllers/welcome_controller_spec.rb]: Its okay to leave TODOS. But I wouldn't leave code that is half done

File: blog/spec/factories/comments.rb [hunk 11]
```diff
@@ -0,0 +1,7 @@
+FactoryGirl.define do
+  factory :comment do
+    # commenter 'somebody'
+    # body 'an interesting comment'
```
Comment (epintos) [inline blog/spec/factories/comments.rb]: It would be nice to use faker here

File: blog/spec/models/comment_spec.rb [hunk 12]
```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  let! (:article) { FactoryGirl.create(:article) }
+  let! (:comment) { Comment.new(commenter: 'somebody',
+                                body: 'an interesting comment',
+                                article: article) }
+  # TODO: The model sholudnt allow comment without articles
+  # @invalid_comment = FactoryGirl.build(:comment, commenter: 'somebody', body: 'an interesting comment', article: nil)
```
Comment (epintos) [inline blog/spec/models/comment_spec.rb]: Same here. It can make the code a bit messy if it has lots of comments of this kind

**Comment by epintos** [comment]
:+1:

**Comment by Enanodr** [comment]
Manually merged to master

**Pull request closed without merging**